tsdb/exemplar.go

```diff
@@ -0,0 +1,33 @@
+package tsdb
+
+import (
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type ExemplarStorage struct {
+	exemplars map[uint64][]exemplar.Exemplar
+}
+
+// NewExemplarStorage creates new in-memory storage for exemplars.
+func NewExemplarStorage() ExemplarStorage {
+	return ExemplarStorage{
+		exemplars: make(map[uint64][]exemplar.Exemplar),
+	}
+}
+
+// Select returns exemplars for a given series labels hash.
```

Comment:
Let's not introduce interfaces that are bound to end up having hash collisions, APIs should be in terms of labels.